Initium-Odp/src/com/universeprojects/miniup/server/ODPDBAccess.java

```diff
@@ -4754,8 +4754,55 @@ else if (currentLocationKey.getId()==pathLocation2Key.getId())
 
 		if (destination.getProperty("ownerKey")!=null)
 		{
-			if (isInParty)
+			if (isInParty) {
+				List<CachedEntity> partyMembers = getParty(db, character);
+				List<CachedEntity> partyGroups = new ArrayList<CachedEntity>();
+				List<CachedEntity> partyUsers = new ArrayList<CachedEntity>();
+				for(CachedEntity partyMember: partyMembers) {
+					//Removes those who have just applied
+					String groupStatus = (String)character.getProperty("groupStatus");
+					if(("Member".equals(groupStatus)==false && "Admin".equals(groupStatus)==false)) {
+						partyGroups.add(new CachedEntity());
```

Comment:
I guess you're creating the entity to avoid a null? If you do GameUtils.equals() calls to check for comparison, you could actually pass it a null and avoid creating new entities that way.